internal/handlers/common/findandmodify.go

```diff
@@ -32,6 +35,13 @@ type FindAndModifyParams struct {
 	MaxTimeMS                             int32
 }
 
+// UpsertParams represents parameters for upsert, if the document exists Update is set.
+// Otherwise, Insert is set. It returns ReturnValue to return to the client.
+type UpsertParams struct {
+	Insert, Update *types.Document
```

Comment:
Hmm, is it possible that both `Insert` and `Update` are not nil?

If not, should this be structured differently?
E.g. 
```
doc *types.Document
insert bool // if false, update operation is expected
```

And also, I don't really understand what is `ReturnValue`, what is the difference between it and the document itself?